cloudinit/util.py

```diff
@@ -1279,6 +1284,26 @@ def find_devs_with_netbsd(criteria=None, oformat='device',
     return []
 
 
+def find_devs_with_openbsd(criteria=None, oformat='device',
+                           tag=None, no_cache=False, path=None):
+    out, _err = subp(['sysctl', '-n', 'hw.disknames'], rcs=[0])
+    devlist = []
+    for entry in out.split(','):
+        if not entry.endswith(':'):
+            # ffs partition with a serial, not a config-drive
+            continue
+        if entry == 'fd0:':
+            continue
+        part_id = 'a' if entry.startswith('cd') else 'i'
+        devlist.append(entry[:-1] + part_id)
+    if criteria:
+        if criteria == "TYPE=iso9660":
+            devlist = [i for i in devlist if i.startswith('cd')]
+        elif criteria == "LABEL=CONFIG-2":
+            devlist = [i for i in devlist if not i.startswith('cd')]
```

Comment:
If we see `criteria` that we don't support, should we do something other than ignore them? (I'm not sure what the expected behaviour here is, this isn't a leading question.)

Reply:
Ok, I added an extra `if` case of that.